worlds/dkc3/Client.py

```diff
@@ -0,0 +1,220 @@
+                    elif num_upgrades == 2:
+                        snes_buffered_write(ctx, WRAM_START + 0x605, bytes([inventory[0] | 0x02]))
+                        if inventory[4] == 0:
+                            snes_buffered_write(ctx, WRAM_START + 0x609, bytes([0x02]))
+                        elif inventory[6] == 0:
+                            snes_buffered_write(ctx, WRAM_START + 0x60B, bytes([0x02]))
+                        elif inventory[8] == 0:
+                            snes_buffered_write(ctx, WRAM_START + 0x60D, bytes([0x02]))
+                        elif inventory[10] == 0:
+                            snes_buffered_write(ctx, WRAM_START + 0x60F, bytes([0x02]))
+                    elif num_upgrades == 3:
+                        snes_buffered_write(ctx, WRAM_START + 0x606, bytes([inventory[1] | 0x20]))
+
+                        k3_ridge_progress = await snes_read(ctx, WRAM_START + 0x693, 0x2)
+                        snes_buffered_write(ctx, WRAM_START + 0x693, bytes([k3_ridge_progress[0] | 0x01]))
+                        snes_buffered_write(ctx, WRAM_START + 0x694, bytes([k3_ridge_progress[1] | 0x01]))
+                elif item.item == 0xDC3000:
+                    # Handle Victory
+                    if not ctx.finished_game:
+                        await ctx.send_msgs([{"cmd": "StatusUpdate", "status": ClientStatus.CLIENT_GOAL}])
+                        ctx.finished_game = True
+                else:
+                    print("Item Not Recognized: ", item.item)
+                pass
+
+            await snes_flush_writes(ctx)
+
+        # DKC3_TODO: This method of collect should work, however it does not unlock the next level correctly when previous is flagged
+        # Handle Collected Locations
+        #for loc_id in ctx.checked_locations:
+        #    if loc_id not in ctx.locations_checked:
+        #        loc_data = location_rom_data[loc_id]
+        #        data = await snes_read(ctx, WRAM_START + loc_data[0], 1)
+        #        invert_bit = ((len(loc_data) >= 3) and loc_data[2])
+        #        if not invert_bit:
+        #            masked_data = data[0] | (1 << loc_data[1])
+        #            print("Collected Location: ", hex(loc_data[0]), " | ", loc_data[1])
+        #            snes_buffered_write(ctx, WRAM_START + loc_data[0], bytes([masked_data]))
+        #            await snes_flush_writes(ctx)
+        #        else:
+        #            masked_data = data[0] & ~(1 << loc_data[1])
+        #            print("Collected Inverted Location: ", hex(loc_data[0]), " | ", loc_data[1])
+        #            snes_buffered_write(ctx, WRAM_START + loc_data[0], bytes([masked_data]))
+        #            await snes_flush_writes(ctx)
+        #        ctx.locations_checked.add(loc_id)
+
+        # Calculate Boomer Cost Text
+        boomer_cost_text = await snes_read(ctx, 0xF5AAFD, 2)
```

Comment:
`0xF5AAFD` should be changed to `WRAM_START + 0xAAFD` for comprehensibility (0xF5 prefix is usb2snes-protocol-specific only, AAFD is the actual memory offset). 3 additional instances in the file of a large integer beginning with "0xF5" that should be split.